# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/extensions/webidl-api/ExtensionEventListener.h
# Commit: 379cd5044768
# Full Hash: 379cd5044768c61ee7b94cc383fa498433a30999
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P1 Make WorkerRunnable to be a base class for runnables on Worker thread and Worker's parent thread. r=dom-worker-reviewers,asuth
#   
#   This is the first step in splitting the parent thread runnable out of WorkerRunnable.
#   
#   To reuse the runnable dispatching codes in Worker, we still need a base class for runnable on the worker thread and the parent thread.
# ==============================================================================

diff -r 29cb10d35b5a -r 379cd5044768 toolkit/components/extensions/webidl-api/ExtensionEventListener.h
--- a/toolkit/components/extensions/webidl-api/ExtensionEventListener.h	Fri Apr 19 09:32:48 2024 +0000
+++ b/toolkit/components/extensions/webidl-api/ExtensionEventListener.h	Fri Apr 19 09:41:57 2024 +0000
@@ -120,7 +120,8 @@
 // A WorkerRunnable subclass used to call an ExtensionEventListener
 // in the thread that owns the dom::Function wrapped by the
 // ExtensionEventListener class.
-class ExtensionListenerCallWorkerRunnable final : public dom::WorkerRunnable {
+class ExtensionListenerCallWorkerRunnable final
+    : public dom::WorkerThreadRunnable {
   friend class ExtensionListenerCallPromiseResultHandler;
 
  public:
@@ -133,8 +134,9 @@
       UniquePtr<dom::StructuredCloneHolder> aArgsHolder,
       ListenerCallOptions* aCallOptions,
       RefPtr<dom::Promise> aPromiseRetval = nullptr)
-      : WorkerRunnable(aExtensionEventListener->GetWorkerPrivate(),
-                       "ExtensionListenerCallWorkerRunnable", WorkerThread),
+      : WorkerThreadRunnable(aExtensionEventListener->GetWorkerPrivate(),
+                             "ExtensionListenerCallWorkerRunnable",
+                             WorkerRunnable::WorkerThread),
         mListener(aExtensionEventListener),
         mArgsHolder(std::move(aArgsHolder)),
         mPromiseResult(std::move(aPromiseRetval)),
